src/features/character-details/DetailsView.tsx

```diff
@@ -0,0 +1,42 @@
+import { type FC } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+import { CharacterDetails } from '@/entities/character/CharacterDetails.tsx';
+import { CloseButton } from '@/shared/close-button/CloseButton.tsx';
+import { ErrorMessage } from '@/shared/error-message/ErrorMessage.tsx';
+import { OverlayWrapper } from '@/shared/overlay-wrapper/OverlayWrapper.tsx';
+import { Spinner } from '@/shared/spinner/Spinner.tsx';
+import { useGetCharacterByIdQuery } from '@/utils/api/rickAndMorty.ts';
+import { useLockBodyScrollOnMobile } from '@/utils/hooks/useLockBodyScrollOnMobile.ts';
+
+export const DetailsView: FC = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const detailsId = searchParams.get('details');
+  useLockBodyScrollOnMobile(!!detailsId);
```

Comment:
detailsId используется дальше в коде как булевое значение, мне кажется было бы логично и более читаемо вынести его в переменную:
```suggestion
  const isDetailsOpen = Boolean(detailsId);
  useLockBodyScrollOnMobile(isDetailsOpen);
```